multiapi-engine/src/main/java/com/sngular/api/generator/plugin/openapi/utils/MapperAuthUtil.java

```diff
@@ -6,52 +6,51 @@
 
+public class MapperAuthUtil {
   private MapperAuthUtil() {}
 
-  public static List<AuthSchemaObject> createAuthSchemaList(final OpenAPI openAPI) {
+  public static List<AuthSchemaObject> createAuthSchemaList(final JsonNode securitySchemes) {
     final ArrayList<AuthSchemaObject> authList = new ArrayList<>();
-    if (Objects.nonNull(openAPI.getComponents()) && MapUtils.isNotEmpty(openAPI.getComponents().getSecuritySchemes())) {
-      openAPI.getComponents().getSecuritySchemes().forEach((key, value) -> {
-        final var typeStr = value.getType().toString();
-        final var isHttpBearer = "http".equalsIgnoreCase(typeStr) && "bearer".equalsIgnoreCase(value.getScheme());
+    if (Objects.nonNull(securitySchemes)) {
+      securitySchemes.fields().forEachRemaining(stringJsonNodeEntry -> {
+        final String nameSecurityScheme = stringJsonNodeEntry.getKey();
+        final String typeStr =getTextValue(stringJsonNodeEntry.getValue(), TYPE);
+        final var scheme = getTextValue(stringJsonNodeEntry.getValue(), SCHEME);
+        final var in = getTextValue(stringJsonNodeEntry.getValue(), IN);
+        final var isHttpBearer = HTTP.equalsIgnoreCase(typeStr) && BEARER.equalsIgnoreCase(scheme);
         final var authSchema = AuthSchemaObject.builder()
-                                               .type(isHttpBearer ? "HttpBearerAuth" : getModelTypeAuth(value)).name(key)
-                                               .apiKeyParam(API_KEY.equalsIgnoreCase(typeStr) ? value.getName() : "")
-                                               .apiKeyPlace(API_KEY.equalsIgnoreCase(typeStr) ? value.getIn().toString() : "")
-                                               .bearerSchema(isHttpBearer ? value.getScheme() : "").build();
+                .type(isHttpBearer ? HTTP_BEARER_AUTH : getModelTypeAuth(typeStr)).name(nameSecurityScheme)
+                .apiKeyParam(API_KEY.equalsIgnoreCase(typeStr) ? nameSecurityScheme: "")
+                .apiKeyPlace(API_KEY.equalsIgnoreCase(typeStr) ? in : "")
+                .bearerSchema(isHttpBearer ? scheme : "").build();
         authList.add(authSchema);
       });
     }
     return authList;
   }
 
-  private static String getModelTypeAuth(final SecurityScheme securityScheme) {
-    var type = securityScheme.getType().toString();
-    if (securityScheme.getType().toString().equalsIgnoreCase(API_KEY)) {
-      type = "ApiKeyAuth";
-    } else if (securityScheme.getType().toString().equalsIgnoreCase("oauth2")) {
-      type = "OAuth";
-    } else if (securityScheme.getType().toString().equalsIgnoreCase("http")) {
-      type = "HttpBasicAuth";
+  private static String getModelTypeAuth(final String type) {
+    if (API_KEY.equalsIgnoreCase(type)) {
+      return API_KEY_AUTH;
+    } else if (type.equalsIgnoreCase(OAUTH_2)) {
+      return StringUtils.capitalize(OAUTH);
+    } else if (type.equalsIgnoreCase(HTTP)) {
+      return HTTP_BASIC_AUTH;
```

Comment:
In these cases we typically store the value to return in a final variable that's uninitialized at the start of the function and set where these returns are, with a single return at the end of the function